### PR TITLE
[BUGFIX] respect descriptor's `configurable` property

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

**Review by rwjblue — COMMENTED** [review]
This seems like a new feature? Where are we using this? What do we need it for?

**Comment by bekzod** [comment]
use case is simple disallow redefining property descriptor so `configurable: false` actually works for example `length` of `ArrayProxy` can be redefined if this PR merged and `length` property defined as `configurable: false` it will become non-redefineable :)    https://github.com/bekzod/ember.js/blob/99561cbbddccb12db41823fad3bb91ce281dcfd8/packages/ember-runtime/lib/system/array_proxy.js#L198
similarly there are several places in codebase that has descriptors  with `configurable: false`
https://github.com/emberjs/ember.js/blob/master/packages/ember-views/lib/mixins/child_views_support.js#L16
https://github.com/emberjs/ember.js/blob/master/packages/ember-views/lib/mixins/view_support.js#L154
also `isDestroyed` and `isDestroying` descriptors could be non configurable too,
https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/system/core_object.js#L263
so they are not redefined when extending object

**Comment by rwjblue** [comment]
@krisselden - Thoughts?

**Review by krisselden — CHANGES_REQUESTED** [review]
This seems like a new feature masquerading as a BUGFIX, this kind of descriptor isn't public as far as I'm aware and we are trying to move to class syntax for get/set props.

File: packages/ember-metal/lib/mixin.ts [hunk 1]
```diff
@@ -379,10 +384,9 @@ function applyMixin(obj: { [key: string]: any }, mixins: Mixin[], partial: boole
   // * Copying `toString` in broken browsers
   mergeMixins(mixins, meta, descs, values, obj, keys);
 
-  for (let i = 0; i < keys.length; i++) {
-    key = keys[i];
+  keys.forEach(key => {
```
[the remark this answers is not in the document]
Comment (bekzod) [reply packages/ember-metal/lib/mixin.ts]: because `keys` is `Set` now

[the remark this answers is not in the document]
Comment (krisselden) [reply packages/ember-metal/lib/mixin.ts]: we can't casually make that change without testing

**Comment by krisselden** [comment]
We don't want to add anything to extending that doesn't align with ES2015 classes.  Property attributes do not inherit from a prototype property.

image

**Pull request closed without merging**